src/neo/SmartContract/Native/Designate/DesignateContract.cs

```diff
@@ -37,6 +37,37 @@ public ECPoint[] GetDesignatedByRole(StoreView snapshot, Role role, uint index)
                 .FirstOrDefault() ?? System.Array.Empty<ECPoint>();
         }
 
+        [ContractMethod(0_01000000, CallFlags.AllowModifyStates)]
+        private byte[] GetDesignatedInfo(StoreView snapshot, Role role, ECPoint node)
+        {
+            var nodes = GetDesignatedByRole(snapshot, role, snapshot.HeaderHeight);
+            if (!nodes.Contains(node))
+                throw new InvalidOperationException("Node not found");
+
+            var value = snapshot.Storages.TryGet(CreateStorageKey((byte)role).Add(node));
+            return value?.Value ?? System.Array.Empty<byte>();
+        }
+
+        [ContractMethod(0_05000000, CallFlags.AllowModifyStates)]
+        private void SetDesignatedInfo(ApplicationEngine engine, Role role, ECPoint node, byte[] value)
+        {
+            if (!engine.CheckWitness(node.EncodePoint(true)))
+                throw new InvalidOperationException("Wrong signed by node");
+
+            var nodes = GetDesignatedByRole(engine.Snapshot, role, engine.Snapshot.HeaderHeight);
+            if (!nodes.Contains(node))
+                throw new InvalidOperationException("Node not found");
+
+            if (value == null || value.Length == 0)
+            {
+                engine.Snapshot.Storages.Delete(CreateStorageKey((byte)role).Add(node));
+            }
+            else
+            {
+                engine.Snapshot.Storages.GetAndChange(CreateStorageKey((byte)role).Add(node), () => new StorageItem(value, false)).Value = value;
+            }
+        }
+
         [ContractMethod(0, CallFlags.AllowModifyStates)]
         private void DesignateAsRole(ApplicationEngine engine, Role role, ECPoint[] nodes)
```

Comment:
```suggestion
        private void DesignateAsRole(ApplicationEngine engine, Role role, ECPoint[] nodes, byte[][] values)
```

Add `values`?  Otherwise, each node has to config it by themselves.

Reply:
Then it must be the committee, and if the oracle change his IP, he will need to the committee agreement.

Reply:
> if the oracle change his IP, he will need to the committee agreement.

He can call `SetDesignatedInfo`.

I mean we need both. When the committee specifies oracle nodes, can also incidentally bind their IP.

Reply:
But if the information it's stored with a key that committee doesn't know?

Reply:
> But if the information it's stored with a key that committee doesn't know?

What do you mean?

Reply:
The committee doesn't need to know the ip address, this encryption key should be agreed between the oracle nodes.

Reply:
There are some operational difficulties. In fact, they are appointed by the committee and need to be coordinated by the committee, so the coordination work between them(like oracle nodes) should be reduced as much as possible.

Reply:
If the key is distributed by the committee, it is better to distribute the configuration file directly by the committee. I think this PR can be closed.

Reply:
This PR require a interaction between the oracles nodes to setup a shared key (the first time), but from this point, they can change the ip address at any time without interact again each other.